src/components/ThemeModeSwitcher/index.jsx

```diff
@@ -0,0 +1,30 @@
+/* Temporary component for testing theme implementation and dark mode.  Used in the right side
+nav menu currently.
+
+@TODO convert to some final implementation
+*/
+import {
+  //   Experimental_CssVarsProvider as CssVarsProvider,
+  useColorScheme,
+} from '@mui/material/styles';
+import styles from './modeSwitcher.module.css';
+
+export const ModeSwitcher = () => {
+  const { mode, setMode } = useColorScheme();
+
+  return (
+    <button
+      variant="outlined"
+      onClick={() => {
+        if (mode === 'light') {
+          setMode('dark');
+        } else {
+          setMode('light');
+        }
+      }}
```

Comment:
This is actually technically using stale data to update `setMode`. The right way to do this kind of conditional update based on previous state in React is with the callback version of `useState`:
```suggestion
      onClick={() => setMode(prevMode => prevMode === 'light' ? 'dark' : 'light')}
```

Reply:
Interesting, this code was from the MUI docs for this feature, so I guess they've got some room for improvement, I'll make that change :slightly_smiling_face:

Reply:
@Centerville1 It's possible that the custom hook doesn't have a callback format.

Your code is also not correct, so that would be the issue if the callback does work. The inner arrow function for the callback doesn't actually return anything. The way that arrow functions work, if you include braces after the arrow, then the function body is a statement that returns nothing unless you use an explicit return statement. Conversely, if you don't use braces for the body of the function, it's an expression, meaning that whatever value the expression evaluates to  is the value returned by the function.

For example, this logs `undefined` because the body of `addOne` is a statement with no `return` statement.
```javascript
const addOne = (x) => { x + 1 };
console.log(addOne(1));
```
Whereas this logs `2`
```javascript
const addOne = (x) => x + 1;
console.log(addOne(1));
```

Reply:
> @Centerville1 It's possible that the custom hook doesn't have a callback format.
> 
> Your code is also not correct, so that would be the issue if the callback does work. The inner arrow function for the callback doesn't actually return anything. The way that arrow functions work, if you include braces after the arrow, then the function body is a statement that returns nothing unless you use an explicit return statement. Conversely, if you don't use braces for the body of the function, it's an expression, meaning that whatever value the expression evaluates to is the value returned by the function.
> 
> For example, this logs `undefined` because the body of `addOne` is a statement with no `return` statement.
> 
> ```js
> const addOne = (x) => { x + 1 };
> console.log(addOne(1));
> ```
> 
> Whereas this logs `2`
> 
> ```js
> const addOne = (x) => x + 1;
> console.log(addOne(1));
> ```

I tried it with and without brackets, I only just added the brackets when it didn't work without them.  I couldn't find anything online about useColorScheme having a callback function, although tbf there isn't that much documentation in general.

Reply:
Gotcha, then it probably doesn't have a callback form, which is fine. The way you had it before isn't awful.

Reply:
Thanks to Amos we were able to do it a little simpler, using ternary operators to do it, so should be a little simpler.